src/hotspot/cpu/ppc/gc/shenandoah/shenandoahBarrierSetAssembler_ppc.cpp

```diff
@@ -599,7 +596,8 @@ void ShenandoahBarrierSetAssembler::store_check(MacroAssembler* masm, Register b
     __ add(base, ind_or_offs.as_register(), base);
   }
 
-  __ load_const_optimized(tmp, (address)ct->byte_map_base(), R0);
+  __ ld(tmp, in_bytes(ShenandoahThreadLocalData::card_table_offset()), R16_thread);
+  __ add(tmp, /* card_table_base */ tmp, R0);
```

Comment:
This looks wrong. R0 is an uninitialized register. We shouldn't add it. (Note that it is used as scratch register in the old code.)